lib/lib.js

```diff
@@ -144,35 +159,42 @@ export function formatDate(d) {
 //     children: { [string]: PageTree },
 //   }
 
-export function buildPageTree(path, type = "", content = false) {
+export function buildPageTree(path, ordering = "", content = false) {
+  // get metadata from _index.md file in each folder. This is where page ordering (weight) is stored.
   const metadata = matter(fs.readFileSync(join(path, "_index.md")), options);
-
+  // get a list of contents at the path, with file types so that the item at the path can be identified as file or folder
   const children = fs.readdirSync(path, { withFileTypes: true });
+  // get a list of files
   const pages = children.filter((f) => f.isFile() && f.name !== "_index.md");
+  // get a list of folders
   const subdirs = children.filter((f) => f.isDirectory());
-
+  // return a pagetree datastructure
   return {
     title: metadata.data.title,
-    pages: pages.map((page) => {
-      const { data: pageData, content: pageContent } = matter(
-        fs.readFileSync(join(path, page.name)),
-        options
-      );
-      return {
-        title: pageData.title,
-        base: page.name,
-        slug: page.name.replace(/.md$/, ""),
-        ...(type === "weight" && { weight: pageData?.weight ?? 0 }),
-        ...(type === "date" && {
-          date: pageData?.date ?? "2000-01-01T00:00:00.000Z",
-        }),
-        ...(content && { content: pageContent }),
-      };
-    }),
+    pages: pages
+      .map((page) => {
+        // retrieve topmatter as JSON
+        const { content, data } = matter(
+          fs.readFileSync(join(path, page.name)),
+          options
+        );
+
+        return {
+          title: data.title,
+          base: page.name,
+          slug: page.name.replace(/.md$/, ""),
+          ...(ordering === "weight" && { weight: data?.weight ?? 0 }),
+          ...(ordering === "date" && {
+            date: data?.date ?? "2000-01-01T00:00:00.000Z",
+          }),
+        };
+      })
+      // Sort files according to weight. Weight can be a date or number. The choice between the two is set by the 'ordering' param.
```

Comment:
Actually not true, this will need a conditional on `type` to change sort algorithm

Reply:
True, missed this